calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -396,12 +447,34 @@ func (w *PeerWatcher) SetBGPConf(bgpConf *calicov3.BGPConfigurationSpec) {
 	w.BGPConf = bgpConf
 }
 
-func NewPeerWatcher(clientv3 calicov3cli.Interface, log *logrus.Entry) *PeerWatcher {
+// Get the BGP password from SecretWatcher
+func (w *PeerWatcher) getPassword(secretKeySelector *v1.SecretKeySelector) (string, error) {
+	password, err := w.secretWatcher.GetSecret(
+		secretKeySelector.Name,
+		secretKeySelector.Key,
+	)
+	return password, err
+}
+
+// This function gets called from SecretWatcher when a secret is added, updated or deleted
+func (w *PeerWatcher) OnSecretUpdate(secret string) {
+	select {
+	case w.secretWatcherEventChan <- secret:
```

Comment:
You can just write
````go
w.secretWatcherEventChan <- secret
````

Reply:
Also if we us the `peerWatcherEventChan` 
this can become 
````go
func OnSecretUpdate(old, new *v1.Secret) {
    common.SendEvent(common.CalicoVppEvent{
        Type: common.BGPSecretChanged,
        Old: old,
        New: new,
    })
}
````
this also needs to add the event type to the filter
````go
reg.ExpectEvents(common.PeerNodeStateChanged, common.BGPSecretChanged)
````

Reply:
> You can just write
> 
> ```go
> w.secretWatcherEventChan <- secret
> ```

Thanks :)